### PR TITLE
feat: add all types

### PR DESCRIPTION
**Pull request by notfilippo**
Support serialisations for all BSON types (http://bsonspec.org/spec.html)

- [x] `Double(f64)`
- [x] `String(String)`
- [x] `Document(Document)`
- [x] `Boolean(bool)`
- [x] `Null`
- [x] `DateTime(chrono::DateTime<Utc>)`
- [x] `Array(Array)`
- [x] `RegularExpression(Regex)`
- [ ] `JavaScriptCode(String)`
- [ ] `JavaScriptCodeWithScope(JavaScriptCodeWithScope)`
- [x] `Int32(i32)`
- [x] `Int64(i64)`
- [x] `Timestamp(Timestamp)`
- [x] `Binary(Binary)`
- [x] `ObjectId(oid::ObjectId)`
- [x] ~~`Decimal128(Decimal128)`~~ not supported in wasm
- [x] `MaxKey`
- [x] `MinKey`

### CODE REVIEW

File: bench/index.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import bson from "bson";
+import { performance } from "perf_hooks";
+
+const date = new Date();
+const data = bson.serialize(
+  {
+    foo: "bar",
+    hello: 55,
+    date,
+    binary: new Buffer.alloc(128),
```
Comment (TTtie) [inline bench/index.mjs]:
```suggestion
    binary: Buffer.alloc(128),
```
Buffer.alloc(...) is not a constructor.

File: bench/index.mjs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const start = performance.now();
+
+for (let i = 0; i < 50000; i++) {
+  const date = new Date();
+  const data = bson.serialize(
+    {
+      foo: "bar",
+      hello: 55,
+      date,
+      binary: new Buffer.alloc(128),
```
Comment (TTtie) [inline bench/index.mjs]:
```suggestion
      binary: Buffer.alloc(128),
```

**Review by TTtie — COMMENTED** [review]
Tiny nitpicks :eyes:

**Review by lucacasonato — APPROVED** [review]
LGTM. Even if this is not final, it gives us something to play with for wire protocol implementation.

**Pull request merged**